Add time entry, include start with create

benthayer opened the pull request:
Hello again,

I added the functionality to add entries with start/stop time. 

Also there was a bug in the parseTime function. I fixed it so that time entries are correct for midnight and then I moved it into the utils file

benthayer committed:
Add time entry, incl start with create

benthayer committed:
Clean up code

beauraines commented:
@benthayer  I've been traveling and am catching up. This is on my todo list, but it will be a few days before i get back to it.

benthayer commented:
> @benthayer  I've been traveling and am catching up. This is on my todo list, but it will be a few days before i get back to it.

All good, no rush. I'm using my local version, so no rush merging this in, just wanted to share my code

benthayer committed:
Fixed bug displaying time of current time entry

beauraines committed:
minor tweak to debug

beauraines commented on cmds/addTimeEntry.mjs:
```diff
@@ -0,0 +1,69 @@
+
+const debug = debugClient('toggl-cli-add');
+
+export const command = 'add <startTime> <endTime> <description>'
```
I think these should be optional using the `[]` syntax, so that the `start`, `end` and `description` arguments can be used, too.

This will give us more flexibility - I think about adding new entries like "I did xxx from start until end", but it wil work in the "From start to end, I did xxx"

I'll make that change and test - but I'm in the middle of making brunch and I discovered that I can push changes!

beauraines replied:
With the change in 

this works

`DEBUG=toggl-cli-add ./cli.js add -d foo -s 8:30 -e '9:02 AM'`

as well as what you had originally 

`DEBUG=toggl-cli-add ./cli.js add 8:00 '08:33 AM' foo`

benthayer replied:
Great, glad we have functionality we both like!

beauraines committed:
works with both positional and option specified values

Makes the positional `startTime`, `endTime` and `description` optional, so that they can _also_ be specified through cli options

beauraines committed:
Removes unused variables and import

beauraines commented on cmds/addTimeEntry.mjs:
```diff
@@ -0,0 +1,68 @@
+  }
+  argv.description ? params.description = argv.description : undefined
+  debug(params)
+  const timeEntry = await client.timeEntries.create(params)
```
I have a `createTimeEntry()` in the utils, but it doesn't support creating a time entry _with an endTime_. I'll open a new issue to extend that function and use it here too.  I see that you started with the `edit` command which does its own interaction with the client. I'm not very good about repeating myself or abstracting methods.

beauraines commented on utils.js:
```diff
@@ -138,3 +142,23 @@ export const displayTimeEntry = async function (timeEntry) {
+ * @param {string} timeString timelike string e.g. 4:50PM '12:00 AM' etc.
+ * @returns {object} dayjs object
+ */
+export function parseTime (timeString) {
```
Thank you for  moving this here, this really should have been available for reuse.
